acceptance/tests/restart_host_run_puppet.rb

```diff
@@ -48,7 +48,14 @@
 
   step "restart each agent" do
     applicable_agents.each do |agent|
-      agent.reboot
+      # Work-around reboot hanging on reply (BKR-971)
+      begin
+        Timeout.timeout(5) do
+          agent.reboot
+        end
```

Comment:
Might need a rescue in there somewhere, just ran across an exception from closing the connection during the reboot call.